src/App.tsx

```diff
@@ -1,11 +1,65 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
+import { getMovie } from './api';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [isMovieFound, setIsMovieFound] = useState(false);
+
+  const resetForm = useCallback(() => {
+    setMovie(null);
+    setQuery('');
+    setIsMovieFound(false);
+    setHasError(false);
+  }, []);
+
+  const handleAddMovie = (movieToAdd: Movie | null) => {
+    if (movieToAdd) {
+      const isAlreadyAdded = movies
+        .find(movieToCheck => movieToCheck.imdbId === movieToAdd.imdbId);
+
+      if (!isAlreadyAdded) {
+        setMovies(prevMovies => [...prevMovies, movieToAdd]);
+      }
+
+      resetForm();
+    }
+  };
+
+  const handleFindMovie: React.FormEventHandler = (event) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    getMovie(query)
+      .then(res => {
+        if ('imdbID' in res) {
+          const foundMovie = {
+            title: res.Title,
+            description: res.Plot,
+            imgUrl: res.Poster !== 'N/A'
+              ? res.Poster
+              : 'https://via.placeholder.com/360x270.png?text=no%20preview',
```

Comment:
```suggestion
            imgUrl: res.Poster || 'https://via.placeholder.com/360x270.png?text=no%20preview',
```

Reply:
>  imgUrl: res.Poster || 'https://via.placeholder.com/360x270.png?text=no%20preview',

this won't work because if there is no poster, the Poster value is a string 'N/A'